# HydraDB Raw API Tests — `sia-test`

Raw-HTTP verification of documented claims (no SDK). Each test maps to a claim in `hydradb_claims_extraction.md` / `api_test_plan.md`.

**How to run:** one cell at a time, top to bottom. Credentials load from `.env` in this folder — nothing secret lives in cells. Each test's markdown states the claim, the source, and what to check in the output. Record the verdict in `api_test_plan.md` as we go.

Base URL: `https://api.hydradb.com` · Auth: `Authorization: Bearer <key>` · `API-Version: 2`

In [3]:
# Legacy cell (superseded by .env — kept for reference)
TENANT = "sia-test"
# HYDRA_DB_API_KEY now lives in .env

In [4]:
%%bash
# Sanity: confirm .env loads and curl exists
set -a; source .env; set +a
echo "base url : $HYDRA_DB_BASE_URL"
echo "tenant   : $HYDRA_DB_TENANT_ID"
echo "key      : ${HYDRA_DB_API_KEY:0:12}... (${#HYDRA_DB_API_KEY} chars)"
curl --version | head -1

base url : https://api.hydradb.com
tenant   : sia-test
key      : sk_test_XglZ... (64 chars)
curl 8.12.1 (Darwin) libcurl/8.12.1 OpenSSL/3.0.18 (SecureTransport) zlib/1.2.13 zstd/1.5.6 libssh2/1.11.1 nghttp2/1.57.0


---
## T01 — No auth → 401 with error envelope

**Claim (API Reference → Error Responses):** every request needs `Authorization: Bearer`; missing/invalid key → `401 UNAUTHORIZED`, and errors use the standard envelope `{success:false, data:null, error:{code,message}, meta:{...}}`.

**Check:** HTTP status is 401; body is the documented envelope with `error.code == "UNAUTHORIZED"`; `meta.request_id` present even on errors.

In [5]:
%%bash
set -a; source .env; set +a
curl -sS "$HYDRA_DB_BASE_URL/databases" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -o /tmp/t01.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
python3 -m json.tool /tmp/t01.json 2>/dev/null || cat /tmp/t01.json

HTTP 403   total=0.456682s
{
    "detail": {
        "error_code": "FORBIDDEN",
        "message": "Missing Authorization header",
        "success": false
    }
}


---
## T02 — Authenticated `GET /databases` → envelope + naming

**Claims:**
- Core endpoints return `{success, data, error, meta}` with `meta.request_id` and `meta.latency_ms` (API Reference index).
- List response uses the migrated name `data.databases[]` (formerly `tenant_ids`) plus `data.failed_databases[]` (List Databases page).
- Side-check: does the account already contain `stock-decoder`? Free tier is claimed to cap at **1 database** ("403 Plan limit reached" — Onboarding cookbook), which determines whether we can create `sia-test` at all.

**Check:** 200; envelope shape; field naming (`databases` vs `tenant_ids`); which databases exist.

In [6]:
%%bash
set -a; source .env; set +a
curl -sS "$HYDRA_DB_BASE_URL/databases" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -o /tmp/t02.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
python3 -m json.tool /tmp/t02.json 2>/dev/null || cat /tmp/t02.json

HTTP 200   total=0.313007s
{
    "success": true,
    "data": {
        "tenant_ids": [
            "default-tenant",
            "sia-test",
            "stock-decoder",
            "stock-market-decoder"
        ],
        "databases": [
            "default-tenant",
            "sia-test",
            "stock-decoder",
            "stock-market-decoder"
        ],
        "failed_tenant_ids": null,
        "failed_databases": null,
        "message": "Successfully retrieved tenant IDs"
    },
    "error": null,
    "meta": {
        "request_id": "4d4d02a3-028f-404a-809a-f52ac92708bc",
        "latency_ms": 50.3
    }
}


---
## T03 — Omit the `API-Version` header

**Claim (API Reference → index / SDKs):** `API-Version: 2` is only "recommended" on raw HTTP (SDKs set it automatically). Docs never say what happens without it — defaulting to v1? v2? Silent?

**Check:** compare status + body to T02. Any difference in shape, any `meta.deprecation[]` entries, any v1-style field names (`tenant_ids`).

In [7]:
%%bash
set -a; source .env; set +a
curl -sS "$HYDRA_DB_BASE_URL/databases" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -o /tmp/t03.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
python3 -m json.tool /tmp/t03.json 2>/dev/null || cat /tmp/t03.json

HTTP 200   total=0.477915s
{
    "success": true,
    "data": {
        "tenant_ids": [
            "default-tenant",
            "sia-test",
            "stock-decoder",
            "stock-market-decoder"
        ],
        "databases": [
            "default-tenant",
            "sia-test",
            "stock-decoder",
            "stock-market-decoder"
        ],
        "failed_tenant_ids": null,
        "failed_databases": null,
        "message": "Successfully retrieved tenant IDs"
    },
    "error": null,
    "meta": {
        "request_id": "bbf59dd7-f707-43d5-bfdc-0926b9ca295a",
        "latency_ms": 44.4
    }
}


---
## T04 — Duplicate `Authorization` headers

**Claim (API Reference → index):** sending two `Authorization` headers is "explicitly called out as a failure mode" — but the docs never say which error it produces. Undocumented behavior on a documented failure mode.

**Check:** status code + error.code. 400? 401? Or does one header silently win (worse)?

In [8]:
%%bash
set -a; source .env; set +a
curl -sS "$HYDRA_DB_BASE_URL/databases" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "Authorization: Bearer duplicate-second-header" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -o /tmp/t04.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
python3 -m json.tool /tmp/t04.json 2>/dev/null || cat /tmp/t04.json

HTTP 403   total=0.298288s
{
    "detail": {
        "error_code": "FORBIDDEN",
        "message": "unauthorized",
        "success": false
    }
}


---
# Phase 1 — Database lifecycle (fresh db: `sia-test-2`)

Everything below targets `sia-test-2` (`HYDRA_DB_TENANT_ID_2` in `.env`) so the retest starts from a clean database. Account context from T02: free tier, already 4 databases.

## T05 — Create `sia-test-2` via `POST /databases`

**Claims:**
- Cookbook-canonical create path is `POST /databases` + `{"database": ...}` (Competitive Intel, Notion AI, Perplexity cookbooks) — while the API Reference's own create page says the path is `/tenants` with `tenant_id` as the only field. T05c tests the other path.
- "Free plan is capped at 1 database — 403 Plan limit reached" (Onboarding cookbook troubleshooting). This account is confirmed free tier and already holds 4 databases, so a success here **kills that claim**.
- Creation is async: expect a 202-style `{status:"accepted"}` and NOT immediately usable (Create Database page).

**Check:** status code (2xx vs 403); response shape (`status:"accepted"`? envelope? which field names echo back).

In [9]:
%%bash
set -a; source .env; set +a
curl -sS -X POST "$HYDRA_DB_BASE_URL/databases" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -H "Content-Type: application/json" \
  -d "{\"database\": \"$HYDRA_DB_TENANT_ID_2\"}" \
  -o /tmp/t05.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
python3 -m json.tool /tmp/t05.json 2>/dev/null || cat /tmp/t05.json

HTTP 200   total=0.362205s
{
    "success": true,
    "data": {
        "status": "accepted",
        "tenant_id": "sia-test-2",
        "database": "sia-test-2",
        "message": "Database creation started in the background. Use GET /databases/status?database=... to check progress."
    },
    "error": null,
    "meta": {
        "request_id": "aed774b4-f05a-4188-9faf-d3db67dd1f04",
        "latency_ms": 52.4
    }
}


## T06 — Poll `GET /databases/status` immediately after create

**Claims:**
- "`TENANT_NOT_FOUND` right after creation is expected transient behavior" (Create Database page) — poll #1 fired immediately should catch it if real.
- `infra.ready_for_ingestion` appears in the docs' example but **not** in their OpenAPI schema property list (schema/prose mismatch) — does it exist?
- `vectorstore_status` is a named-keys object (`.knowledge`/`.memories`), not the v1 array.

**Check:** first-poll behavior; presence/shape of `ready_for_ingestion` and `vectorstore_status`; how long until ready.

In [10]:
%%bash
set -a; source .env; set +a
for i in 1 2 3 4 5 6; do
  echo "--- poll $i ---"
  curl -sS "$HYDRA_DB_BASE_URL/databases/status?database=$HYDRA_DB_TENANT_ID_2" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
    -o /tmp/t06.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
  python3 -m json.tool /tmp/t06.json 2>/dev/null || cat /tmp/t06.json
  grep -q '"ready_for_ingestion": true' /tmp/t06.json && break
  sleep 5
done

--- poll 1 ---
HTTP 200   total=0.319093s
{
    "success": true,
    "data": {
        "tenant_id": "sia-test-2",
        "database": "sia-test-2",
        "org_id": "5b6a2ub3ts",
        "infra": {
            "scheduler_status": true,
            "graph_status": true,
            "vectorstore_status": {
                "knowledge": true,
                "memories": false
            },
            "ready_for_ingestion": false
        },
        "message": "Deployed infrastructure status"
    },
    "error": null,
    "meta": {
        "request_id": "7bff833a-d67c-4828-885f-312687ca2e99",
        "latency_ms": 56.2
    }
}
--- poll 2 ---
HTTP 200   total=0.321004s
{
    "success": true,
    "data": {
        "tenant_id": "sia-test-2",
        "database": "sia-test-2",
        "org_id": "5b6a2ub3ts",
        "infra": {
            "scheduler_status": true,
            "graph_status": true,
            "vectorstore_status": {
                "knowledge": true,
                "memories"

## T05b — Duplicate create via `/databases` → the 409 claim

**Claims:**
- Error taxonomy documents `409 TENANT_ALREADY_EXISTS` for re-creating an existing database.
- T01 showed the *auth* error path ignores the documented envelope — does an application-level error use `{success,data,error,meta}` as documented, or the `{"detail":{...}}` shape again?

**Check:** 409 vs something else; error code name; envelope vs detail-wrapper; `meta.request_id` present on error.

In [11]:
%%bash
set -a; source .env; set +a
curl -sS -X POST "$HYDRA_DB_BASE_URL/databases" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -H "Content-Type: application/json" \
  -d "{\"database\": \"$HYDRA_DB_TENANT_ID_2\"}" \
  -o /tmp/t05b.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
python3 -m json.tool /tmp/t05b.json 2>/dev/null || cat /tmp/t05b.json

HTTP 409   total=0.329545s
{
    "detail": {
        "success": false,
        "message": "Database ID already exists",
        "error_code": "DATABASE_ALREADY_EXISTS"
    }
}


## T05c — Does `POST /tenants` (the API Reference's own path) exist?

**Claim:** the API Reference "Create Database" page documents `POST /tenants` with `tenant_id` as the only field (no `database` alias) — conflicting with every cookbook. Since `sia-test-2` now exists, a real endpoint should answer 409; a nonexistent path answers 404/405.

**Check:** does the path exist at all, and if so which field names does it accept?

In [12]:
%%bash
set -a; source .env; set +a
curl -sS -X POST "$HYDRA_DB_BASE_URL/tenants" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -H "Content-Type: application/json" \
  -d "{\"tenant_id\": \"$HYDRA_DB_TENANT_ID_2\"}" \
  -o /tmp/t05c.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
python3 -m json.tool /tmp/t05c.json 2>/dev/null || cat /tmp/t05c.json

HTTP 409   total=0.303000s
{
    "detail": {
        "success": false,
        "message": "Organisation tenant ID already exists",
        "error_code": "INVALID_INPUT"
    }
}


## T07 — `stats` + `collections` on the empty database

**Claims:**
- `row_count` counts **chunks, not source documents** (Database Stats page, flagged as a "common mistake") — baseline should be 0 now; re-check after Phase 2 ingest.
- Stats response: `{database, knowledge_collection:{row_count,dimensions}, memory_collection:{...}}`.
- Collections response includes the deprecated `sub_tenant_ids` alias alongside `collections` (List Collections page); a default collection is auto-provisioned at creation.

**Check:** both shapes; whether a default collection already exists; dual-naming duplication like T02.

In [13]:
%%bash
set -a; source .env; set +a
echo "== GET /databases/stats =="
curl -sS "$HYDRA_DB_BASE_URL/databases/stats?database=$HYDRA_DB_TENANT_ID_2" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -o /tmp/t07a.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
python3 -m json.tool /tmp/t07a.json 2>/dev/null || cat /tmp/t07a.json
echo
echo "== GET /databases/collections =="
curl -sS "$HYDRA_DB_BASE_URL/databases/collections?database=$HYDRA_DB_TENANT_ID_2" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -o /tmp/t07b.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
python3 -m json.tool /tmp/t07b.json 2>/dev/null || cat /tmp/t07b.json

== GET /databases/stats ==
HTTP 200   total=0.325649s
{
    "success": true,
    "data": {
        "tenant_id": "sia-test-2",
        "database": "sia-test-2",
        "knowledge_collection": {
            "row_count": 0,
            "dimensions": 1536
        },
        "memory_collection": {
            "row_count": 0,
            "dimensions": 1536
        },
        "message": "Successfully retrieved tenant collection statistics"
    },
    "error": null,
    "meta": {
        "request_id": "45f6bb96-bd95-499e-8c5f-ccc464a3e3cb",
        "latency_ms": 55.3
    }
}

== GET /databases/collections ==
HTTP 200   total=0.300942s
{
    "success": true,
    "data": {
        "sub_tenant_ids": [],
        "collections": [],
        "message": "Successfully retrieved sub-tenant IDs"
    },
    "error": null,
    "meta": {
        "request_id": "06dd0d33-ec90-4c97-b62a-af2793d4e9d8",
        "latency_ms": 43.2
    }
}


---
# Phase 2 — Ingestion (the two Atlas PDFs → `sia-test-2`)

Run T08 → T14 in order, **without long pauses between T09 and T13** (T13 deliberately queries before indexing finishes).

## T08 — Raw JSON body to `/context/ingest` → the "SDK required" claim

**Claim (Customer Support + Financial Analyst cookbooks):** raw HTTP calls with a JSON body to `/context/ingest` return `422`; the documented "fix" is *use the official SDK* (rather than documenting the multipart shape).

**Check:** 422? What does the error say — does it explain multipart is needed, or leak internals (SDK-era testing saw server-side field name `file_metadata` leak in errors)?

In [14]:
%%bash
set -a; source .env; set +a
curl -sS -X POST "$HYDRA_DB_BASE_URL/context/ingest" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -H "Content-Type: application/json" \
  -d "{\"database\": \"$HYDRA_DB_TENANT_ID_2\", \"type\": \"knowledge\", \"documents\": [\"placeholder\"]}" \
  -o /tmp/t08.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
python3 -m json.tool /tmp/t08.json 2>/dev/null || cat /tmp/t08.json

HTTP 400   total=0.506091s
{
    "detail": {
        "success": false,
        "message": "database is required",
        "error_code": "INVALID_INPUT"
    }
}


## T09 — Correct multipart ingest of the RFC PDF

**Claims:** field is `documents` (not `files`), scope field `database`; returns `202 Accepted` (async); response `{success_count, failed_count, results:[{id, filename, status, error}]}` — some cookbooks show a bare `{results:[...]}` without counts. Also: T05 returned 200 where docs implied 202 — does ingest?

**Check:** status code; response wrapper shape; capture the returned id → `/tmp/hydra_rfc_id` for later tests.

In [15]:
%%bash
set -a; source .env; set +a
curl -sS -X POST "$HYDRA_DB_BASE_URL/context/ingest" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -F "database=$HYDRA_DB_TENANT_ID_2" \
  -F "type=knowledge" \
  -F "documents=@rfc_project_atlas_migration.pdf" \
  -o /tmp/t09.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
python3 -m json.tool /tmp/t09.json 2>/dev/null || cat /tmp/t09.json
python3 - << 'EOF'
import json
d = json.load(open('/tmp/t09.json'))
body = d.get('data', d)
results = body.get('results') or []
if results and results[0].get('id'):
    open('/tmp/hydra_rfc_id','w').write(results[0]['id'])
    print('captured rfc id:', results[0]['id'])
else:
    print('!! no id found — inspect response above')
EOF

HTTP 202   total=0.511848s
{
    "success": true,
    "data": {
        "success": true,
        "message": "Knowledge uploaded successfully",
        "results": [
            {
                "id": "074fa243a53f8956221a020f6013e05d",
                "filename": "rfc_project_atlas_migration.pdf",
                "status": "queued",
                "error": null,
                "error_code": null
            }
        ],
        "success_count": 1,
        "failed_count": 0
    },
    "error": null,
    "meta": {
        "request_id": "7c1d059c-7949-4ba6-b1df-1df983073c85",
        "latency_ms": 253.3,
        "tenant_id": "sia-test-2",
        "database": "sia-test-2",
        "sub_tenant_id": "7qtkangrs6",
        "collection": "7qtkangrs6",
        "source_type": "knowledge"
    }
}
captured rfc id: 074fa243a53f8956221a020f6013e05d


## T10 — Wrong multipart field name (`files=`)

**Claim:** the Cursor-for-Docs cookbook contradicts itself, calling the field `files` in one section and `documents` in its own troubleshooting fix. What actually happens with `files=`?

**Check:** clear 400/422, or something silent/confusing? (A silent 2xx with nothing ingested would be the worst outcome — note `success_count` if present.)

In [16]:
%%bash
set -a; source .env; set +a
curl -sS -X POST "$HYDRA_DB_BASE_URL/context/ingest" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -F "database=$HYDRA_DB_TENANT_ID_2" \
  -F "type=knowledge" \
  -F "files=@rfc_project_atlas_migration.pdf" \
  -o /tmp/t10.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
python3 -m json.tool /tmp/t10.json 2>/dev/null || cat /tmp/t10.json

HTTP 400   total=0.325356s
{
    "detail": {
        "success": false,
        "message": "Provide at least one of 'files' (documents) or 'app_knowledge' (JSON).",
        "error_code": "INVALID_INPUT"
    }
}


## T11 — `document_metadata` shape quirks (meeting-notes PDF)

**Claims (SDK-era findings, re-verified raw):**
1. A **single object** (not array-wrapped) for one document errors — and the error text leaks the server-side field name `file_metadata`.
2. A **structurally-wrong-but-valid-JSON** shape (array, but items missing the required `id`/`metadata` wrapper) is **silently accepted** with success, leaving chunk metadata permanently empty.
3. Correct shape: array of `{id, metadata}` objects.

Call (a) sends the single object (expect error). Call (b) sends the wrong-but-valid array shape (per the claim, expect silent success — this doc's metadata will stay empty; that's deliberate, we check it in Phase 4). Captures the meeting-notes id.

**Check:** (a)'s status + whether `file_metadata` leaks; (b)'s status — silent acceptance confirmed?

In [17]:
%%bash
set -a; source .env; set +a
echo "== (a) single object — expect error =="
curl -sS -X POST "$HYDRA_DB_BASE_URL/context/ingest" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -F "database=$HYDRA_DB_TENANT_ID_2" \
  -F "type=knowledge" \
  -F "documents=@meeting_notes_atlas_rollback.pdf" \
  -F 'document_metadata={"doc_kind": "meeting_notes", "project": "atlas"}' \
  -o /tmp/t11a.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
python3 -m json.tool /tmp/t11a.json 2>/dev/null || cat /tmp/t11a.json
echo
echo "== (b) wrong-but-valid array shape — claim says silent success =="
curl -sS -X POST "$HYDRA_DB_BASE_URL/context/ingest" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -F "database=$HYDRA_DB_TENANT_ID_2" \
  -F "type=knowledge" \
  -F "documents=@meeting_notes_atlas_rollback.pdf" \
  -F 'document_metadata=[{"doc_kind": "meeting_notes", "project": "atlas"}]' \
  -o /tmp/t11b.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
python3 -m json.tool /tmp/t11b.json 2>/dev/null || cat /tmp/t11b.json
python3 - << 'EOF'
import json
try:
    d = json.load(open('/tmp/t11b.json'))
    body = d.get('data', d)
    results = body.get('results') or []
    if results and results[0].get('id'):
        open('/tmp/hydra_meeting_id','w').write(results[0]['id'])
        print('captured meeting id:', results[0]['id'])
except Exception as e:
    print('parse failed:', e)
EOF

== (a) single object — expect error ==
HTTP 400   total=0.344488s
{
    "detail": {
        "success": false,
        "message": "Invalid file_metadata: json: cannot unmarshal object into Go value of type []json.RawMessage",
        "error_code": "INVALID_INPUT"
    }
}

== (b) wrong-but-valid array shape — claim says silent success ==
HTTP 202   total=0.499513s
{
    "success": true,
    "data": {
        "success": true,
        "message": "Knowledge uploaded successfully",
        "results": [
            {
                "id": "5af80448e53b663acfb46c65dc0ee118",
                "filename": "meeting_notes_atlas_rollback.pdf",
                "status": "queued",
                "error": null,
                "error_code": null
            }
        ],
        "success_count": 1,
        "failed_count": 0
    },
    "error": null,
    "meta": {
        "request_id": "77eb9cd0-a0d6-479c-96da-1a73285529ad",
        "latency_ms": 245.5,
        "tenant_id": "sia-test-2",
        "databa

## T12 — `/context/status`: `database=` vs `tenant_id=` param conflict

**Claim conflict:** the API Reference says this endpoint takes **`tenant_id` only** (no `database` alias); the cookbooks all call it with `database=`. Also watch for the undocumented `success` enum value alongside the 5 documented states (`queued/processing/graph_creation/completed/errored`).

**Check:** which param works (both? one? — each call below uses ONLY one); status values seen.

In [18]:
%%bash
set -a; source .env; set +a
IDS="$(cat /tmp/hydra_rfc_id 2>/dev/null),$(cat /tmp/hydra_meeting_id 2>/dev/null)"
echo "ids: $IDS"
echo "== (a) database= only =="
curl -sS "$HYDRA_DB_BASE_URL/context/status?database=$HYDRA_DB_TENANT_ID_2&ids=$IDS" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -o /tmp/t12a.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
python3 -m json.tool /tmp/t12a.json 2>/dev/null || cat /tmp/t12a.json
echo
echo "== (b) tenant_id= only =="
curl -sS "$HYDRA_DB_BASE_URL/context/status?tenant_id=$HYDRA_DB_TENANT_ID_2&ids=$IDS" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -o /tmp/t12b.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
python3 -m json.tool /tmp/t12b.json 2>/dev/null || cat /tmp/t12b.json

ids: 074fa243a53f8956221a020f6013e05d,5af80448e53b663acfb46c65dc0ee118
== (a) database= only ==
HTTP 200   total=0.327652s
{
    "success": true,
    "data": {
        "statuses": [
            {
                "id": "074fa243a53f8956221a020f6013e05d,5af80448e53b663acfb46c65dc0ee118",
                "indexing_status": "errored",
                "error_code": "FILE_NOT_FOUND",
                "error_message": "",
                "success": false,
                "message": "ID not found"
            }
        ]
    },
    "error": null,
    "meta": {
        "request_id": "d6f33339-5f55-4f93-8241-256987b214dd",
        "latency_ms": 59.9,
        "tenant_id": "sia-test-2",
        "database": "sia-test-2",
        "sub_tenant_id": "7qtkangrs6",
        "collection": "7qtkangrs6"
    }
}

== (b) tenant_id= only ==
HTTP 200   total=0.314270s
{
    "success": true,
    "data": {
        "statuses": [
            {
                "id": "074fa243a53f8956221a020f6013e05d,5af80448e53b663acf

## T13 — Query immediately (indexing race) — run this RIGHT AFTER T12

**Claim (4+ cookbooks, self-admitted):** "Querying immediately returns empty results with no error, which looks like a bug but is not… wait 30 seconds and retry." There is no "not yet indexed" signal — just an empty `chunks[]`.

**Check:** 200 + empty chunks with zero indication why? Or has this improved? (If T12 already showed `completed`, this test is moot — note that.)

In [19]:
%%bash
set -a; source .env; set +a
curl -sS -X POST "$HYDRA_DB_BASE_URL/query" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -H "Content-Type: application/json" \
  -d "{\"tenant_id\": \"$HYDRA_DB_TENANT_ID_2\", \"query\": \"What is Project Atlas?\", \"max_results\": 5}" \
  -o /tmp/t13.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
python3 -m json.tool /tmp/t13.json 2>/dev/null || cat /tmp/t13.json

HTTP 200   total=0.814518s
{
    "success": true,
    "data": {
        "chunks": [],
        "sources": [],
        "graph_context": {
            "query_paths": [],
            "chunk_relations": [],
            "chunk_id_to_group_ids": {},
            "synthesis_context": null
        },
        "additional_context": {}
    },
    "error": null,
    "meta": {
        "request_id": "84b6c9ee-02d8-4192-9729-8de3e2e7d325",
        "latency_ms": 537.9,
        "tenant_id": "sia-test-2",
        "database": "sia-test-2",
        "sub_tenant_id": "7qtkangrs6",
        "collection": "7qtkangrs6",
        "source_type": "knowledge",
        "deprecation": [
            {
                "deprecated": true,
                "message": "The tenant_id and sub_tenant_id fields are deprecated. Migrate to the database and collection fields.",
                "deprecated_since": "2.0.1"
            }
        ]
    }
}


## T14 — `/context/status` with a fabricated id

**Claim (Ingestion Status page):** "unknown IDs return `errored` rather than being dropped" — i.e. you cannot distinguish a typo'd id from a real ingestion failure.

**Check:** does a nonsense id really come back as `errored`? (If so, that's a footgun worth writing up — a typo looks identical to a real failure.)

In [20]:
%%bash
set -a; source .env; set +a
curl -sS "$HYDRA_DB_BASE_URL/context/status?database=$HYDRA_DB_TENANT_ID_2&ids=totally-fake-id-12345" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -o /tmp/t14.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
python3 -m json.tool /tmp/t14.json 2>/dev/null || cat /tmp/t14.json

HTTP 200   total=0.348047s
{
    "success": true,
    "data": {
        "statuses": [
            {
                "id": "totally-fake-id-12345",
                "indexing_status": "errored",
                "error_code": "FILE_NOT_FOUND",
                "error_message": "",
                "success": false,
                "message": "ID not found"
            }
        ]
    },
    "error": null,
    "meta": {
        "request_id": "81120ae4-a9a4-40fc-9b46-3ca4eed98903",
        "latency_ms": 81.4,
        "tenant_id": "sia-test-2",
        "database": "sia-test-2",
        "sub_tenant_id": "7qtkangrs6",
        "collection": "7qtkangrs6"
    }
}


---
*Phase 2 ends here. Wait ~1–2 min after T13, then re-run the T12 (a) cell once more to see the final indexing state (`graph_creation` → `completed`) before we move to Phase 3 (query & retrieval — the core entity-resolution and temporal claims).*

## T12-fix — repeated `ids=` params (comma-separated was treated as one id)

T12 revealed `ids=a,b` gets swallowed as a single id. Retrying with `ids=a&ids=b` (the OpenAPI `ids[]` style). Also shows the real indexing lifecycle for our two docs — run this until both show `completed` (docs say `graph_creation` is already queryable).

**Check:** per-id splitting works this way?; status values (`queued/processing/graph_creation/completed` — or the undocumented `success`?).

In [21]:
%%bash
set -a; source .env; set +a
RFC=$(cat /tmp/hydra_rfc_id); MTG=$(cat /tmp/hydra_meeting_id)
curl -sS "$HYDRA_DB_BASE_URL/context/status?database=$HYDRA_DB_TENANT_ID_2&ids=$RFC&ids=$MTG" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -o /tmp/t12fix.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
python3 -m json.tool /tmp/t12fix.json 2>/dev/null || cat /tmp/t12fix.json

HTTP 200   total=0.351921s
{
    "success": true,
    "data": {
        "statuses": [
            {
                "id": "074fa243a53f8956221a020f6013e05d",
                "indexing_status": "completed",
                "error_code": "",
                "error_message": "",
                "success": true,
                "message": "Processing status retrieved successfully"
            },
            {
                "id": "5af80448e53b663acfb46c65dc0ee118",
                "indexing_status": "completed",
                "error_code": "",
                "error_message": "",
                "success": true,
                "message": "Processing status retrieved successfully"
            }
        ]
    },
    "error": null,
    "meta": {
        "request_id": "ef5092fa-839c-4e96-ba9c-c36a8f826d94",
        "latency_ms": 54.2,
        "tenant_id": "sia-test-2",
        "database": "sia-test-2",
        "sub_tenant_id": "7qtkangrs6",
        "collection": "7qtkangrs6"
    }
}


## T07-recheck — does the auto-created collection now appear?

T09's `meta` exposed an auto-assigned collection `7qtkangrs6` (opaque id, not "default"). Does `GET /databases/collections` list it now that content exists — i.e. are collections "implicitly created on first write" (docs claim) and *listed* once used?

**Check:** `collections`/`sub_tenant_ids` arrays now non-empty? Same opaque id?

In [22]:
%%bash
set -a; source .env; set +a
curl -sS "$HYDRA_DB_BASE_URL/databases/collections?database=$HYDRA_DB_TENANT_ID_2" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -o /tmp/t07r.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
python3 -m json.tool /tmp/t07r.json 2>/dev/null || cat /tmp/t07r.json
echo
echo "== stats recheck (row_count should now be chunk-count, not 2) =="
curl -sS "$HYDRA_DB_BASE_URL/databases/stats?database=$HYDRA_DB_TENANT_ID_2" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -o /tmp/t07r2.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
python3 -m json.tool /tmp/t07r2.json 2>/dev/null || cat /tmp/t07r2.json

HTTP 200   total=0.336389s
{
    "success": true,
    "data": {
        "sub_tenant_ids": [
            "7qtkangrs6"
        ],
        "collections": [
            "7qtkangrs6"
        ],
        "message": "Successfully retrieved sub-tenant IDs"
    },
    "error": null,
    "meta": {
        "request_id": "537dc3df-2ba2-48e3-b7a9-4847598eb266",
        "latency_ms": 58
    }
}

== stats recheck (row_count should now be chunk-count, not 2) ==
HTTP 200   total=0.375797s
{
    "success": true,
    "data": {
        "tenant_id": "sia-test-2",
        "database": "sia-test-2",
        "knowledge_collection": {
            "row_count": 2,
            "dimensions": 1536
        },
        "memory_collection": {
            "row_count": 0,
            "dimensions": 1536
        },
        "message": "Successfully retrieved tenant collection statistics"
    },
    "error": null,
    "meta": {
        "request_id": "938c8452-bbb4-44be-b141-2e1712f5a5c1",
        "latency_ms": 66.5
    }
}


---
*Run T12-fix (repeat until both docs are `completed`), then T07-recheck. Save + "done" — Phase 3 (query & retrieval: entity resolution, temporal reasoning, the fast/thinking contradiction) comes next.*

---
# Phase 3 — Query & retrieval (the core claims)

Both docs are `completed`, so retrieval + graph should be fully available. The first cell writes a small response summarizer to `/tmp/hydra_sum.py` used by later cells (full JSON where shape matters, summaries where ranking matters).

In [23]:
%%bash
cat > /tmp/hydra_sum.py << 'EOF'
import json, sys
d = json.load(open(sys.argv[1]))
data = d.get('data') or {}
ch = data.get('chunks') or []
print(f"chunks: {len(ch)}")
for c in ch[:8]:
    txt = (c.get('chunk_content') or '')[:100].replace('\n',' ')
    print(f"  [{c.get('relevancy_score')}] {c.get('source_title')} :: {txt}")
gc = data.get('graph_context') or {}
rel = gc.get('chunk_relations') or []
print('chunk_relations:', len(rel), '| query_paths:', len(gc.get('query_paths') or []))
sc = gc.get('synthesis_context')
print('synthesis_context:', 'None' if sc is None else str(sc)[:400])
m = d.get('meta') or {}
print('meta.latency_ms:', m.get('latency_ms'))
EOF
echo summarizer written

summarizer written


## T15 — `/query`: `database=` vs `tenant_id=`

**Claim (API Reference):** `/query` accepts `tenant_id` only, "no `database` alias at all" — on the single most-used endpoint. T13 already caught a deprecation notice when using `tenant_id`, implying the docs have it backwards.

**Check:** does `database`-only work? Does `tenant_id`-only still work + deprecation notice? Same results both ways?

In [24]:
%%bash
set -a; source .env; set +a
echo "== (a) database= only =="
curl -sS -X POST "$HYDRA_DB_BASE_URL/query" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -H "Content-Type: application/json" \
  -d "{\"database\": \"$HYDRA_DB_TENANT_ID_2\", \"query\": \"What is Project Atlas?\", \"max_results\": 5}" \
  -o /tmp/t15a.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
python3 /tmp/hydra_sum.py /tmp/t15a.json
echo
echo "== (b) tenant_id= only =="
curl -sS -X POST "$HYDRA_DB_BASE_URL/query" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -H "Content-Type: application/json" \
  -d "{\"tenant_id\": \"$HYDRA_DB_TENANT_ID_2\", \"query\": \"What is Project Atlas?\", \"max_results\": 5}" \
  -o /tmp/t15b.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
python3 /tmp/hydra_sum.py /tmp/t15b.json
python3 -c "import json; d=json.load(open('/tmp/t15b.json')); print('deprecation:', json.dumps(d.get('meta',{}).get('deprecation')))" 

== (a) database= only ==
HTTP 200   total=0.989247s
chunks: 2
  [0.7371318867722023] rfc_project_atlas_migration.pdf :: ## RFC-019: Migrate Project Atlas to Vector Database  Author: Wallace Okafor (Data Science)   |   Da
  [0.7073258090978621] meeting_notes_atlas_rollback.pdf :: ## Meeting Notes: Project Atlas Post-Migration Review  Date: April 22, 2026   |   Attendees: Priya C
chunk_relations: 15 | query_paths: 10
synthesis_context: None
meta.latency_ms: 488.9

== (b) tenant_id= only ==
HTTP 200   total=0.712729s
chunks: 2
  [0.7371318867722023] rfc_project_atlas_migration.pdf :: ## RFC-019: Migrate Project Atlas to Vector Database  Author: Wallace Okafor (Data Science)   |   Da
  [0.7073258090978621] meeting_notes_atlas_rollback.pdf :: ## Meeting Notes: Project Atlas Post-Migration Review  Date: April 22, 2026   |   Attendees: Priya C
chunk_relations: 15 | query_paths: 10
synthesis_context: None
meta.latency_ms: 376.3
deprecation: [{"deprecated": true, "message": "The tenant_id and s

## T16 — Baseline hybrid, `mode=fast` — full response shape

**Claims:** `chunks[].chunk_content/.source_title/.relevancy_score/.chunk_uuid` (cookbook-documented shape); `graph_context` defaults **true** (param omitted here); `additional_metadata` may carry the undocumented `_retrieval_source` key (SDK-era finding).

**Check:** full JSON — field names as documented? graph fields populated by default? any undocumented keys?

In [25]:
%%bash
set -a; source .env; set +a
curl -sS -X POST "$HYDRA_DB_BASE_URL/query" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -H "Content-Type: application/json" \
  -d "{\"database\": \"$HYDRA_DB_TENANT_ID_2\", \"query\": \"Why was Project Atlas migrated to a vector database?\", \"max_results\": 5, \"mode\": \"fast\"}" \
  -o /tmp/t16.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
python3 -m json.tool /tmp/t16.json 2>/dev/null || cat /tmp/t16.json

HTTP 200   total=0.806062s
{
    "success": true,
    "data": {
        "chunks": [
            {
                "chunk_uuid": "5af80448e53b663acfb46c65dc0ee118_chunk_0000",
                "id": "5af80448e53b663acfb46c65dc0ee118",
                "chunk_content": "## Meeting Notes: Project Atlas Post-Migration Review\n\nDate: April 22, 2026   |   Attendees: Priya Chandrasekaran, Marcus Webb, Wallace Okafor, Alex Kim (Security)\n\n## Context\n\nSix weeks after migrating Project Atlas's similarity search to the vector database per RFC-019, the team reconvened to review production results before the final Postgres decommission step.\n\n## Findings\n\nQuery latency did improve as expected (p95 down to 45ms). However, Alex Kim flagged a security incident during the review period: the row-level security policies that previously gated item visibility b user permissions were not being enforced in the new vector database layer, since it has no native concept of Postgres RLS. Two users briefly

## T17 — `mode=thinking`: `synthesis_context` gating + latency

**Claims:** `synthesis_context` is only populated in `mode:"thinking"` (and only for multi-step/`requires_synthesis` queries) — a gating condition buried in one OpenAPI block; sub-200ms retrieval (server `latency_ms` vs wall clock, thinking mode expected slower).

**Check:** `synthesis_context` non-null here vs null in T16? latency comparison.

In [26]:
%%bash
set -a; source .env; set +a
curl -sS -X POST "$HYDRA_DB_BASE_URL/query" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -H "Content-Type: application/json" \
  -d "{\"database\": \"$HYDRA_DB_TENANT_ID_2\", \"query\": \"Why did the team change its decision about decommissioning Postgres for Project Atlas?\", \"max_results\": 5, \"mode\": \"thinking\"}" \
  -o /tmp/t17.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
python3 /tmp/hydra_sum.py /tmp/t17.json

HTTP 200   total=2.256643s
chunks: 2
  [0.8654120898551545] meeting_notes_atlas_rollback.pdf :: ## Meeting Notes: Project Atlas Post-Migration Review  Date: April 22, 2026   |   Attendees: Priya C
  [0.771117226882845] rfc_project_atlas_migration.pdf :: ## RFC-019: Migrate Project Atlas to Vector Database  Author: Wallace Okafor (Data Science)   |   Da
chunk_relations: 15 | query_paths: 2
synthesis_context: The provided search results identify a "Project Atlas Post-Migration Review" meeting held on April 22, 2026, as the relevant source for this decision. However, the snippets are truncated and do not contain the specific justifications or reasons why the team changed its decision regarding the decommissioning of Postgres.
meta.latency_ms: 1913.9


## T18 — `query_forceful_relations` + `mode=fast`: the docs' self-contradiction

**Claim conflict:** Memories page says sending it with `mode:"fast"` returns **HTTP 422**; Knowledge and Query pages say it's **silently ignored**. Both cannot be true.

**Check:** 422 or 200? If 200 — any warning anywhere (meta?) that the param did nothing?

In [27]:
%%bash
set -a; source .env; set +a
curl -sS -X POST "$HYDRA_DB_BASE_URL/query" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -H "Content-Type: application/json" \
  -d "{\"database\": \"$HYDRA_DB_TENANT_ID_2\", \"query\": \"What is Project Atlas?\", \"max_results\": 3, \"mode\": \"fast\", \"query_forceful_relations\": true}" \
  -o /tmp/t18.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
python3 /tmp/hydra_sum.py /tmp/t18.json
python3 -c "import json; d=json.load(open('/tmp/t18.json')); print('meta:', json.dumps(d.get('meta'), indent=1))" 

HTTP 200   total=0.729482s
chunks: 2
  [0.7371318867722023] rfc_project_atlas_migration.pdf :: ## RFC-019: Migrate Project Atlas to Vector Database  Author: Wallace Okafor (Data Science)   |   Da
  [0.7073258090978621] meeting_notes_atlas_rollback.pdf :: ## Meeting Notes: Project Atlas Post-Migration Review  Date: April 22, 2026   |   Attendees: Priya C
chunk_relations: 15 | query_paths: 10
synthesis_context: None
meta.latency_ms: 385.6
meta: {
 "request_id": "45978090-0bb9-4b60-b7ad-a1c7099e5aed",
 "latency_ms": 385.6,
 "tenant_id": "sia-test-2",
 "database": "sia-test-2",
 "sub_tenant_id": "7qtkangrs6",
 "collection": "7qtkangrs6",
 "source_type": "knowledge"
}


## T19 — Entity resolution across the two documents

**Claim (4 cookbooks):** the context graph "automatically connects" the same entities across sources with no manual tagging. Wallace Okafor authored RFC-019 (doc 1) AND owns the re-filtering follow-up (doc 2); Alex Kim, Marcus Webb, Priya Chandrasekaran also span both.

**Check:** do `graph_context.chunk_relations` actually link entities across the two PDFs? Full relations dump below.

In [28]:
%%bash
set -a; source .env; set +a
curl -sS -X POST "$HYDRA_DB_BASE_URL/query" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -H "Content-Type: application/json" \
  -d "{\"database\": \"$HYDRA_DB_TENANT_ID_2\", \"query\": \"Who is Wallace Okafor and what is he responsible for?\", \"max_results\": 6, \"mode\": \"thinking\", \"graph_context\": true}" \
  -o /tmp/t19.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
python3 /tmp/hydra_sum.py /tmp/t19.json
echo "---- full chunk_relations ----"
python3 -c "
import json
d = json.load(open('/tmp/t19.json'))
gc = (d.get('data') or {}).get('graph_context') or {}
print(json.dumps(gc.get('chunk_relations'), indent=1))
print('query_paths:', json.dumps(gc.get('query_paths'), indent=1))" 

HTTP 200   total=2.154511s
chunks: 2
  [0.6273390829223843] rfc_project_atlas_migration.pdf :: ## RFC-019: Migrate Project Atlas to Vector Database  Author: Wallace Okafor (Data Science)   |   Da
  [0.606552466396229] meeting_notes_atlas_rollback.pdf :: ## Meeting Notes: Project Atlas Post-Migration Review  Date: April 22, 2026   |   Attendees: Priya C
chunk_relations: 15 | query_paths: 10
synthesis_context: The query seeks specific identity and role information about a particular individual, requiring a lookup of a person's profile and responsibilities.
meta.latency_ms: 1709.4
---- full chunk_relations ----
[
 {
  "triplets": [
   {
    "source": {
     "entity_id": "4ca843243fa84fe45769b2b07812d89b",
     "identifier": null,
     "name": "wallace okafor",
     "namespace": "employees",
     "type": "PERSON"
    },
    "relation": {
     "canonical_predicate": "assigned to implement",
     "chunk_id": "5af80448e53b663acfb46c65dc0ee118_chunk_0000",
     "context": "Wallace Okafor is re

## T20 — Knowledge update / temporal reasoning: the superseded decision

**Claims:** LongMemEval Knowledge Update 97.43%, Temporal Reasoning 90.97%; bitemporal edges; "time-aware memory." The RFC (2026-03-04) says decommission Postgres; the meeting notes (2026-04-22) **reverse** that. A correct answer must lead with the reversal.

**Check:** does the top-ranked content reflect the April reversal, or does the RFC's confident (outdated) plan win on vocabulary similarity?

In [29]:
%%bash
set -a; source .env; set +a
curl -sS -X POST "$HYDRA_DB_BASE_URL/query" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -H "Content-Type: application/json" \
  -d "{\"database\": \"$HYDRA_DB_TENANT_ID_2\", \"query\": \"Is Postgres being decommissioned for Project Atlas? What is the current plan?\", \"max_results\": 6, \"mode\": \"thinking\"}" \
  -o /tmp/t20.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
python3 /tmp/hydra_sum.py /tmp/t20.json

HTTP 200   total=1.782601s
chunks: 2
  [0.6656587616344207] rfc_project_atlas_migration.pdf :: ## RFC-019: Migrate Project Atlas to Vector Database  Author: Wallace Okafor (Data Science)   |   Da
  [0.6607671418368644] meeting_notes_atlas_rollback.pdf :: ## Meeting Notes: Project Atlas Post-Migration Review  Date: April 22, 2026   |   Attendees: Priya C
chunk_relations: 15 | query_paths: 2
synthesis_context: The user is asking for a specific status (decommissioning) and a future plan regarding a specific technology (Postgres) within a specific project (Project Atlas). This requires factual lookup of project documentation or roadmaps.
meta.latency_ms: 1337.2


## T21 — `recency_bias` 0.0 vs 0.9

**Claim (Financial Analyst cookbook):** `recency_bias` structurally fixes the "same-vocabulary" failure — two near-identical-vocabulary docs, the newer should win with bias on. Note: both docs were *ingested* minutes apart; whether recency uses document dates (2026-03-04 vs 2026-04-22) or ingest time is itself the question.

**Check:** ranking difference between the two calls; which timestamp notion applies.

In [30]:
%%bash
set -a; source .env; set +a
echo "== recency_bias: 0.0 =="
curl -sS -X POST "$HYDRA_DB_BASE_URL/query" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -H "Content-Type: application/json" \
  -d "{\"database\": \"$HYDRA_DB_TENANT_ID_2\", \"query\": \"What is the plan for Postgres in Project Atlas?\", \"max_results\": 4, \"recency_bias\": 0.0}" \
  -o /tmp/t21a.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
python3 /tmp/hydra_sum.py /tmp/t21a.json
echo
echo "== recency_bias: 0.9 =="
curl -sS -X POST "$HYDRA_DB_BASE_URL/query" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -H "Content-Type: application/json" \
  -d "{\"database\": \"$HYDRA_DB_TENANT_ID_2\", \"query\": \"What is the plan for Postgres in Project Atlas?\", \"max_results\": 4, \"recency_bias\": 0.9}" \
  -o /tmp/t21b.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
python3 /tmp/hydra_sum.py /tmp/t21b.json

== recency_bias: 0.0 ==
HTTP 200   total=0.783227s
chunks: 2
  [0.7702453548040235] rfc_project_atlas_migration.pdf :: ## RFC-019: Migrate Project Atlas to Vector Database  Author: Wallace Okafor (Data Science)   |   Da
  [0.7604393332279373] meeting_notes_atlas_rollback.pdf :: ## Meeting Notes: Project Atlas Post-Migration Review  Date: April 22, 2026   |   Attendees: Priya C
chunk_relations: 15 | query_paths: 2
synthesis_context: None
meta.latency_ms: 442.3

== recency_bias: 0.9 ==
HTTP 200   total=1.139391s
chunks: 2
  [0.7702453548040235] rfc_project_atlas_migration.pdf :: ## RFC-019: Migrate Project Atlas to Vector Database  Author: Wallace Okafor (Data Science)   |   Da
  [0.7604393332279373] meeting_notes_atlas_rollback.pdf :: ## Meeting Notes: Project Atlas Post-Migration Review  Date: April 22, 2026   |   Attendees: Priya C
chunk_relations: 15 | query_paths: 2
synthesis_context: None
meta.latency_ms: 754.5


## T22 — `metadata_filters`: undeclared key silently ignored?

**Claim (List Context + Query pages):** undeclared/non-schema keys are **silently ignored, not errored**. We declared no metadata schema at creation, so `doc_kind` is undeclared — per the claim, results should be identical to an unfiltered call (a filter that does nothing, with no warning).

**Check:** same chunk count as T15a? Any error/warning at all?

In [31]:
%%bash
set -a; source .env; set +a
curl -sS -X POST "$HYDRA_DB_BASE_URL/query" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -H "Content-Type: application/json" \
  -d "{\"database\": \"$HYDRA_DB_TENANT_ID_2\", \"query\": \"What is Project Atlas?\", \"max_results\": 5, \"metadata_filters\": {\"doc_kind\": \"meeting_notes\"}}" \
  -o /tmp/t22.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
python3 /tmp/hydra_sum.py /tmp/t22.json

HTTP 200   total=0.497312s
chunks: 0
chunk_relations: 0 | query_paths: 0
synthesis_context: None
meta.latency_ms: 226.5


## T23 — Zero-result query

**Claim (Query page):** zero matches → `200` with empty arrays, never an error. (Given T13, an off-topic query on a 2-doc corpus should return empty — or does hybrid search return *something* anyway, scored low? Also interesting.)

In [32]:
%%bash
set -a; source .env; set +a
curl -sS -X POST "$HYDRA_DB_BASE_URL/query" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -H "Content-Type: application/json" \
  -d "{\"database\": \"$HYDRA_DB_TENANT_ID_2\", \"query\": \"best chocolate cake recipes with buttercream frosting\", \"max_results\": 5}" \
  -o /tmp/t23.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
python3 /tmp/hydra_sum.py /tmp/t23.json

HTTP 200   total=1.379262s
chunks: 2
  [0.6434689218779721] rfc_project_atlas_migration.pdf :: ## RFC-019: Migrate Project Atlas to Vector Database  Author: Wallace Okafor (Data Science)   |   Da
  [0.6412642033145581] meeting_notes_atlas_rollback.pdf :: ## Meeting Notes: Project Atlas Post-Migration Review  Date: April 22, 2026   |   Attendees: Priya C
chunk_relations: 15 | query_paths: 0
synthesis_context: None
meta.latency_ms: 971.2


## T24 — `max_results: 51` (documented max 50)

**Claim:** default 10, max 50. Over-max is undocumented: 4xx, clamp, or accept?

In [33]:
%%bash
set -a; source .env; set +a
curl -sS -X POST "$HYDRA_DB_BASE_URL/query" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -H "Content-Type: application/json" \
  -d "{\"database\": \"$HYDRA_DB_TENANT_ID_2\", \"query\": \"Project Atlas\", \"max_results\": 51}" \
  -o /tmp/t24.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
python3 /tmp/hydra_sum.py /tmp/t24.json 2>/dev/null || python3 -m json.tool /tmp/t24.json

HTTP 400   total=0.317540s
chunks: 0
chunk_relations: 0 | query_paths: 0
synthesis_context: None
meta.latency_ms: None


## T25 — `alpha` edge cases

**Claims:** `alpha` is ignored when `query_by:"text"` (no error); `alpha:"auto"` is a documented accepted value (float-or-"auto" union).

**Check:** (a) text + alpha → 200, no complaint? (b) `"auto"` accepted on hybrid?

In [34]:
%%bash
set -a; source .env; set +a
echo "== (a) query_by=text + alpha=0.2 (should be ignored) =="
curl -sS -X POST "$HYDRA_DB_BASE_URL/query" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -H "Content-Type: application/json" \
  -d "{\"database\": \"$HYDRA_DB_TENANT_ID_2\", \"query\": \"Postgres\", \"query_by\": \"text\", \"alpha\": 0.2, \"max_results\": 3}" \
  -o /tmp/t25a.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
python3 /tmp/hydra_sum.py /tmp/t25a.json
echo
echo "== (b) hybrid + alpha=\"auto\" =="
curl -sS -X POST "$HYDRA_DB_BASE_URL/query" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -H "Content-Type: application/json" \
  -d "{\"database\": \"$HYDRA_DB_TENANT_ID_2\", \"query\": \"Postgres\", \"alpha\": \"auto\", \"max_results\": 3}" \
  -o /tmp/t25b.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
python3 /tmp/hydra_sum.py /tmp/t25b.json 2>/dev/null || python3 -m json.tool /tmp/t25b.json

== (a) query_by=text + alpha=0.2 (should be ignored) ==
HTTP 200   total=0.321119s
chunks: 2
  [1] rfc_project_atlas_migration.pdf :: ## RFC-019: Migrate Project Atlas to Vector Database  Author: Wallace Okafor (Data Science)   |   Da
  [1] meeting_notes_atlas_rollback.pdf :: ## Meeting Notes: Project Atlas Post-Migration Review  Date: April 22, 2026   |   Attendees: Priya C
chunk_relations: 0 | query_paths: 0
synthesis_context: None
meta.latency_ms: 66.4

== (b) hybrid + alpha="auto" ==
HTTP 200   total=1.036556s
chunks: 2
  [0.6989106692060618] meeting_notes_atlas_rollback.pdf :: ## Meeting Notes: Project Atlas Post-Migration Review  Date: April 22, 2026   |   Attendees: Priya C
  [0.6940431038920056] rfc_project_atlas_migration.pdf :: ## RFC-019: Migrate Project Atlas to Vector Database  Author: Wallace Okafor (Data Science)   |   Da
chunk_relations: 15 | query_paths: 10
synthesis_context: None
meta.latency_ms: 633.3


## T26 — `query_by:"text"` + `operator` (v1 boolean_recall's v2 replacement)

**Claim:** exact/boolean matching moved to `POST /query` with `query_by:"text"` + `operator: phrase|and|or` — and unlike v1's special endpoint, it shouldn't reject other params.

**Check:** phrase match on "permission re-filtering" (wording unique to the meeting notes) hits the right doc?

In [35]:
%%bash
set -a; source .env; set +a
curl -sS -X POST "$HYDRA_DB_BASE_URL/query" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -H "Content-Type: application/json" \
  -d "{\"database\": \"$HYDRA_DB_TENANT_ID_2\", \"query\": \"permission re-filtering\", \"query_by\": \"text\", \"operator\": \"phrase\", \"max_results\": 3}" \
  -o /tmp/t26.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
python3 /tmp/hydra_sum.py /tmp/t26.json

HTTP 200   total=0.325308s
chunks: 1
  [1] meeting_notes_atlas_rollback.pdf :: ## Meeting Notes: Project Atlas Post-Migration Review  Date: April 22, 2026   |   Attendees: Priya C
chunk_relations: 0 | query_paths: 0
synthesis_context: None
meta.latency_ms: 62.5


---
# Phase 4 — Inspect / relations / list / metadata patch

Non-destructive. Run in order. (Phase 5 below is destructive — run it only after Phase 4 is done.)

## T27 — `/context/inspect`: param naming + `expiry_seconds` bounds

**Claims:** this endpoint shows "no `database`/`collection` migration at all" — `tenant_id` required, no alias (API Reference); `expiry_seconds` valid range 60–604800; `mode` ∈ `content|url|both`.

**Check:** (a) `database=` — rejected or accepted (T15 suggests docs may be backwards again)? (b) `tenant_id=` + `expiry_seconds=59` — bounds error or silent acceptance?

In [36]:
%%bash
set -a; source .env; set +a
RFC=$(cat /tmp/hydra_rfc_id)
echo "== (a) database= only, mode=content =="
curl -sS "$HYDRA_DB_BASE_URL/context/inspect?id=$RFC&database=$HYDRA_DB_TENANT_ID_2&mode=content" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -o /tmp/t27a.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
head -c 700 /tmp/t27a.json; echo
echo
echo "== (b) tenant_id= + expiry_seconds=59 + mode=url =="
curl -sS "$HYDRA_DB_BASE_URL/context/inspect?id=$RFC&tenant_id=$HYDRA_DB_TENANT_ID_2&mode=url&expiry_seconds=59" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -o /tmp/t27b.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
python3 -m json.tool /tmp/t27b.json 2>/dev/null | head -40 || cat /tmp/t27b.json

== (a) database= only, mode=content ==
HTTP 200   total=0.846142s
{"success":true,"data":{"success":true,"id":"074fa243a53f8956221a020f6013e05d","content":null,"content_base64":"JVBERi0xLjQKJZOMi54gUmVwb3J0TGFiIEdlbmVyYXRlZCBQREYgZG9jdW1lbnQgKG9wZW5zb3VyY2UpCjEgMCBvYmoKPDwKL0YxIDIgMCBSIC9GMiAzIDAgUgo+PgplbmRvYmoKMiAwIG9iago8PAovQmFzZUZvbnQgL0hlbHZldGljYSAvRW5jb2RpbmcgL1dpbkFuc2lFbmNvZGluZyAvTmFtZSAvRjEgL1N1YnR5cGUgL1R5cGUxIC9UeXBlIC9Gb250Cj4+CmVuZG9iagozIDAgb2JqCjw8Ci9CYXNlRm9udCAvSGVsdmV0aWNhLUJvbGQgL0VuY29kaW5nIC9XaW5BbnNpRW5jb2RpbmcgL05hbWUgL0YyIC9TdWJ0eXBlIC9UeXBlMSAvVHlwZSAvRm9udAo+PgplbmRvYmoKNCAwIG9iago8PAovQ29udGVudHMgOCAwIFIgL01lZGlhQm94IFsgMCAwIDYxMiA3OTIgXSAvUGFyZW50IDcgMCBSIC9SZXNvdXJjZXMgPDwKL0ZvbnQgMSAwIFIgL1Byb2NTZXQgWyAvUERGIC9UZXh0IC9JbWFn

== (b) tenant_id= + expiry_seconds=59 + mode=url ==
HTTP 400   total=0.323687s
{
    "detail": {
        "success": false,
        "message": "expiry_seconds must be between 60 and 604800",
        "error_code": "INVALID_INPUT"
    

## T28 — `/context/relations`: the "opaque" cursor

**Claims:** `cursor` is described as an opaque number but is "literally a relevancy-score float reused as a pagination token"; `limit` 1–10000 default 5000; `tenant_id`-only per docs (test `database=`).

**Check:** cursor value in the response — float? Does `database=` work? Relations shape vs what `/query` returned in T19?

In [37]:
%%bash
set -a; source .env; set +a
curl -sS "$HYDRA_DB_BASE_URL/context/relations?database=$HYDRA_DB_TENANT_ID_2&limit=5" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -o /tmp/t28.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
python3 - << 'EOF'
import json
d = json.load(open('/tmp/t28.json'))
data = d.get('data') or {}
print('top-level data keys:', list(data.keys()))
rels = data.get('relations') or data.get('triplets') or []
print('relations returned:', len(rels) if isinstance(rels, list) else type(rels))
if isinstance(rels, list) and rels:
    print(json.dumps(rels[0], indent=1)[:800])
for k in ('cursor','next_cursor','pagination'):
    if k in data: print(k, '=', json.dumps(data[k]))
print('meta:', json.dumps((d.get('meta') or {}), indent=1)[:300])
EOF

HTTP 200   total=0.312238s
top-level data keys: ['relations', 'is_truncated', 'next_cursor', 'success', 'message']
relations returned: 11
{
 "source": {
  "name": "project atlas",
  "type": "PROJECT",
  "namespace": "projects",
  "entity_id": "632cad1e5b82f227c4c98f8e054fc70e",
  "identifier": null,
  "provider": ""
 },
 "target": {
  "name": "postgres",
  "type": "TECHNOLOGY",
  "namespace": "technologies",
  "entity_id": "c80e66e46fae3b8bbd2f42eb0c64f22b",
  "identifier": null,
  "provider": ""
 },
 "relations": [
  {
   "canonical_predicate": "runs on",
   "raw_predicate": "runs on",
   "context": "Project Atlas currently runs on a single Postgres instance that handles both relational data and similarity search via a hand-rolled cosine-distance query.",
   "confidence": 0.8,
   "temporal_details": null,
   "timestamp": "2026-07-14T04:30:26Z",
   "relationship_id": "72a7069f195c43a90238fd5333b3031e",
   "chunk_id": "074fa2
next_cursor = null
meta: {
 "request_id": "26c17f57-5b43-4caa

## T29 — `/context/list`: caps, forbidden `include_fields`, and the T11b metadata damage

**Claims:** `page_size` 1–100 (test 101); `include_fields` rejects `content`/`url`/`attachments` with 400 ("use /context/inspect"); body is `tenant_id`-only per docs (test `database=`); metadata fields in list output should show the meeting-notes doc's **empty** metadata (T11b damage) vs whatever the RFC has.

**Check:** (a) page_size 101 → 400 or clamp? (b) include_fields ["content"] → 400? (c) normal list — metadata state per doc.

In [38]:
%%bash
set -a; source .env; set +a
echo "== (a) page_size=101 =="
curl -sS -X POST "$HYDRA_DB_BASE_URL/context/list" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -H "Content-Type: application/json" \
  -d "{\"database\": \"$HYDRA_DB_TENANT_ID_2\", \"type\": \"knowledge\", \"page\": 1, \"page_size\": 101}" \
  -o /tmp/t29a.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
head -c 400 /tmp/t29a.json; echo
echo
echo "== (b) include_fields=[content] =="
curl -sS -X POST "$HYDRA_DB_BASE_URL/context/list" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -H "Content-Type: application/json" \
  -d "{\"database\": \"$HYDRA_DB_TENANT_ID_2\", \"type\": \"knowledge\", \"include_fields\": [\"content\"]}" \
  -o /tmp/t29b.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
head -c 400 /tmp/t29b.json; echo
echo
echo "== (c) plain list =="
curl -sS -X POST "$HYDRA_DB_BASE_URL/context/list" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -H "Content-Type: application/json" \
  -d "{\"database\": \"$HYDRA_DB_TENANT_ID_2\", \"type\": \"knowledge\"}" \
  -o /tmp/t29c.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
python3 -m json.tool /tmp/t29c.json 2>/dev/null || cat /tmp/t29c.json

== (a) page_size=101 ==
HTTP 400   total=0.331440s
{"detail":{"success":false,"message":"page_size must be between 1 and 100. See https://docs.hydradb.com/api-reference/v2/endpoint/list-documents for usage details. ","error_code":"INVALID_INPUT"}}

== (b) include_fields=[content] ==
HTTP 400   total=0.290368s
{"detail":{"success":false,"message":"invalid include_fields: content. See https://docs.hydradb.com/api-reference/v2/endpoint/list-documents for usage details. ","error_code":"INVALID_INPUT"}}

== (c) plain list ==
HTTP 200   total=0.562692s
{
    "success": true,
    "data": {
        "success": true,
        "message": "Successfully fetched sources",
        "sources": [
            {
                "additional_metadata": {},
                "app_external_id": null,
                "app_kind": null,
                "app_provider": null,
                "description": "",
                "id": "074fa243a53f8956221a020f6013e05d",
                "metadata": {},
                "n

## T30 — `PATCH /context/sources/{id}/metadata`: naming mess + repairing T11b

**Claims:** request takes `database`/`collection` (collection **required**, no default fallback — omitting it should error); metadata goes in `tenant_metadata` (NOT `metadata` — `document_metadata`/`metadata` are "explicitly rejected"); the **response** reverts to `tenant_id`/`sub_tenant_id` naming — inconsistent within one endpoint.

Two calls: (a) without `collection` → error claim; (b) with `collection=7qtkangrs6` + `additional_metadata` — attempts to repair the meeting-notes doc's empty metadata from T11b.

**Check:** (a) errors how? (b) success + response field naming; then re-list to confirm the repair stuck.

In [39]:
%%bash
set -a; source .env; set +a
MTG=$(cat /tmp/hydra_meeting_id)
echo "== (a) no collection — claim: required, no default fallback =="
curl -sS -X PATCH "$HYDRA_DB_BASE_URL/context/sources/$MTG/metadata" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -H "Content-Type: application/json" \
  -d "{\"database\": \"$HYDRA_DB_TENANT_ID_2\", \"additional_metadata\": {\"doc_kind\": \"meeting_notes\"}}" \
  -o /tmp/t30a.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
head -c 400 /tmp/t30a.json; echo
echo
echo "== (b) with collection — repair T11b metadata =="
curl -sS -X PATCH "$HYDRA_DB_BASE_URL/context/sources/$MTG/metadata" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -H "Content-Type: application/json" \
  -d "{\"database\": \"$HYDRA_DB_TENANT_ID_2\", \"collection\": \"7qtkangrs6\", \"additional_metadata\": {\"doc_kind\": \"meeting_notes\", \"doc_date\": \"2026-04-22\"}}" \
  -o /tmp/t30b.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
python3 -m json.tool /tmp/t30b.json 2>/dev/null || cat /tmp/t30b.json
echo
echo "== (c) verify via /context/list =="
curl -sS -X POST "$HYDRA_DB_BASE_URL/context/list" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -H "Content-Type: application/json" \
  -d "{\"database\": \"$HYDRA_DB_TENANT_ID_2\", \"type\": \"knowledge\", \"ids\": [\"$MTG\"]}" \
  -o /tmp/t30c.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
python3 -m json.tool /tmp/t30c.json 2>/dev/null || cat /tmp/t30c.json

== (a) no collection — claim: required, no default fallback ==
HTTP 400   total=0.305683s
{"detail":{"success":false,"message":"collection is required","error_code":"INVALID_INPUT"}}

== (b) with collection — repair T11b metadata ==
HTTP 200   total=0.752521s
{
    "success": true,
    "data": {
        "id": "5af80448e53b663acfb46c65dc0ee118",
        "tenant_id": "sia-test-2",
        "sub_tenant_id": "7qtkangrs6",
        "updated": true,
        "database_metadata_keys": [],
        "tenant_metadata_keys": [],
        "additional_metadata_keys": [
            "doc_date",
            "doc_kind"
        ],
        "milvus_sync_required": false,
        "chunk_rows_matched": 1,
        "chunk_rows_modified": 1
    },
    "error": null,
    "meta": {
        "request_id": "ea97812c-e702-4828-a665-fae1522452f2",
        "latency_ms": 443.3,
        "tenant_id": "sia-test-2",
        "database": "sia-test-2",
        "deprecation": [
            {
                "deprecated": true,
    

---
# Phase 5 — Delete semantics (DESTRUCTIVE — run last, in order)

Deletes the RFC doc from `sia-test-2`, verifies read-path removal, re-ingests the same file to test the silent-failure claim. `sia-test-2` itself is NOT deleted (T34 targets a nonexistent name).

## T31 — `DELETE /context`: uniquely-nested body + per-id results

**Claims:** body is `{type, request:{database, collection, ids[]}}` — the only endpoint nesting scope under `request` (everything else is flat); knowledge deletes report **per-id** results.

**Check:** does the nested shape work? Per-id `deleted` results? What if you use the flat shape everyone would try first? (a) flat → expected error; (b) nested → works.

In [40]:
%%bash
set -a; source .env; set +a
RFC=$(cat /tmp/hydra_rfc_id)
echo "== (a) flat shape (what everyone tries first) =="
curl -sS -X DELETE "$HYDRA_DB_BASE_URL/context" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -H "Content-Type: application/json" \
  -d "{\"type\": \"knowledge\", \"database\": \"$HYDRA_DB_TENANT_ID_2\", \"ids\": [\"$RFC\"]}" \
  -o /tmp/t31a.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
head -c 400 /tmp/t31a.json; echo
echo
echo "== (b) documented nested shape =="
curl -sS -X DELETE "$HYDRA_DB_BASE_URL/context" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -H "Content-Type: application/json" \
  -d "{\"type\": \"knowledge\", \"request\": {\"database\": \"$HYDRA_DB_TENANT_ID_2\", \"ids\": [\"$RFC\"]}}" \
  -o /tmp/t31b.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
python3 -m json.tool /tmp/t31b.json 2>/dev/null || cat /tmp/t31b.json

== (a) flat shape (what everyone tries first) ==
HTTP 200   total=1.580150s
{"success":true,"data":{"success":true,"message":"Successfully deleted 1 source(s)","results":[{"id":"074fa243a53f8956221a020f6013e05d","deleted":true}],"deleted_count":1},"error":null,"meta":{"request_id":"3e3c40df-c3cc-4031-8e45-1057ea9fca5e","latency_ms":1205.4,"tenant_id":"sia-test-2","database":"sia-test-2","sub_tenant_id":"7qtkangrs6","collection":"7qtkangrs6","source_type":"knowledge"}}

== (b) documented nested shape ==
HTTP 400   total=0.302715s
{
    "detail": {
        "error_code": "INVALID_INPUT",
        "message": "database is required. Pass the 'database' field in the request body.",
        "success": false
    }
}


## T32 — Immediate read-path check after delete

**Claim:** deleted ids vanish from `/query` and `/context/list` **immediately**, even before background cleanup finishes.

**Check:** query mentioning RFC content — is the RFC chunk gone? List — only the meeting-notes doc left? (Run right after T31.)

In [41]:
%%bash
set -a; source .env; set +a
echo "== /query =="
curl -sS -X POST "$HYDRA_DB_BASE_URL/query" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -H "Content-Type: application/json" \
  -d "{\"database\": \"$HYDRA_DB_TENANT_ID_2\", \"query\": \"Who authored RFC-019 and what did it propose?\", \"max_results\": 5}" \
  -o /tmp/t32a.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
python3 /tmp/hydra_sum.py /tmp/t32a.json
echo
echo "== /context/list =="
curl -sS -X POST "$HYDRA_DB_BASE_URL/context/list" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -H "Content-Type: application/json" \
  -d "{\"database\": \"$HYDRA_DB_TENANT_ID_2\", \"type\": \"knowledge\"}" \
  -o /tmp/t32b.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
python3 -c "
import json
d = json.load(open('/tmp/t32b.json'))
data = d.get('data') or {}
docs = data.get('documents') or data.get('sources') or data.get('results') or []
print('docs listed:', len(docs) if isinstance(docs,list) else list(data.keys()))
for x in (docs if isinstance(docs,list) else [])[:5]:
    print(' -', x.get('id'), x.get('filename') or x.get('title'))" 

== /query ==
HTTP 200   total=0.807095s
chunks: 1
  [0.7415059705083361] meeting_notes_atlas_rollback.pdf :: ## Meeting Notes: Project Atlas Post-Migration Review  Date: April 22, 2026   |   Attendees: Priya C
chunk_relations: 11 | query_paths: 5
synthesis_context: None
meta.latency_ms: 457.6

== /context/list ==
HTTP 200   total=0.326624s
docs listed: 1
 - 5af80448e53b663acfb46c65dc0ee118 meeting_notes_atlas_rollback.pdf


## T33 — Re-ingest the just-deleted file: the silent-failure claim

**Claim (SDK-era finding, Significant):** re-ingesting under a just-deleted filename/id returns `success: true` with the same deterministic id, but the document **never actually appears** — a silent failure distinct from E6001. Workaround was "never reuse a just-deleted filename."

**Check:** (a) re-ingest returns 202 + same id as before (`/tmp/hydra_rfc_id`)? (b) poll status ~60s — does it ever reach `completed`? (c) does it show up in `/context/list`? Give this one time.

In [42]:
%%bash
set -a; source .env; set +a
OLD=$(cat /tmp/hydra_rfc_id)
echo "old id: $OLD"
curl -sS -X POST "$HYDRA_DB_BASE_URL/context/ingest" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -F "database=$HYDRA_DB_TENANT_ID_2" \
  -F "type=knowledge" \
  -F "documents=@rfc_project_atlas_migration.pdf" \
  -o /tmp/t33.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
python3 -m json.tool /tmp/t33.json 2>/dev/null || cat /tmp/t33.json
NEW=$(python3 -c "import json; d=json.load(open('/tmp/t33.json')); r=(d.get('data') or d).get('results') or [{}]; print(r[0].get('id',''))")
echo "new id: $NEW   (same as old? $([ \"$NEW\" = \"$OLD\" ] && echo YES || echo no))"
for i in 1 2 3 4 5 6; do
  sleep 10
  curl -sS "$HYDRA_DB_BASE_URL/context/status?database=$HYDRA_DB_TENANT_ID_2&ids=$NEW" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
    -o /tmp/t33s.json -w '' 
  python3 -c "
import json
d = json.load(open('/tmp/t33s.json'))
s = ((d.get('data') or {}).get('statuses') or [{}])[0]
print('poll', $i if False else '', s.get('indexing_status'), s.get('error_code',''))" 
  grep -q '"indexing_status": "completed"' /tmp/t33s.json && break
done
echo "== list check =="
curl -sS -X POST "$HYDRA_DB_BASE_URL/context/list" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -H "Content-Type: application/json" \
  -d "{\"database\": \"$HYDRA_DB_TENANT_ID_2\", \"type\": \"knowledge\"}" \
  -o /tmp/t33l.json -w 'HTTP %{http_code}\n'
python3 -c "
import json
d = json.load(open('/tmp/t33l.json'))
data = d.get('data') or {}
docs = data.get('documents') or data.get('sources') or data.get('results') or []
print('docs listed:', len(docs) if isinstance(docs,list) else list(data.keys()))
for x in (docs if isinstance(docs,list) else [])[:5]:
    print(' -', x.get('id'), x.get('filename') or x.get('title'))" 

old id: 074fa243a53f8956221a020f6013e05d
HTTP 202   total=0.484611s
{
    "success": true,
    "data": {
        "success": true,
        "message": "Knowledge uploaded successfully",
        "results": [
            {
                "id": "074fa243a53f8956221a020f6013e05d",
                "filename": "rfc_project_atlas_migration.pdf",
                "status": "queued",
                "error": null,
                "error_code": null
            }
        ],
        "success_count": 1,
        "failed_count": 0
    },
    "error": null,
    "meta": {
        "request_id": "b7fb614e-72cd-4875-8d76-2c2873a5159f",
        "latency_ms": 227.6,
        "tenant_id": "sia-test-2",
        "database": "sia-test-2",
        "sub_tenant_id": "7qtkangrs6",
        "collection": "7qtkangrs6",
        "source_type": "knowledge"
    }
}
new id: 074fa243a53f8956221a020f6013e05d   (same as old? YES)
poll  graph_creation 
poll  graph_creation 
poll  completed 
poll  completed 
poll  completed 
poll

## T34 — `DELETE /tenants` on a nonexistent database

**Claims:** retrying delete on an already-deleted/nonexistent tenant → `404 TENANT_NOT_FOUND`; deletion is async ("deletion_scheduled"). Targets a name that never existed — nothing real is deleted.

**Check:** 404? Which error code name (docs' `TENANT_NOT_FOUND` vs the `DATABASE_*` drift seen in T05b)? Envelope or detail-wrapper?

In [43]:
%%bash
set -a; source .env; set +a
curl -sS -X DELETE "$HYDRA_DB_BASE_URL/tenants?tenant_id=never-existed-xyz-999" \
  -H "Authorization: Bearer $HYDRA_DB_API_KEY" \
  -H "API-Version: $HYDRA_DB_API_VERSION" \
  -o /tmp/t34.json -w 'HTTP %{http_code}   total=%{time_total}s\n'
python3 -m json.tool /tmp/t34.json 2>/dev/null || cat /tmp/t34.json

HTTP 404   total=0.313759s
{
    "detail": {
        "success": false,
        "message": "tenant never-existed-xyz-999 does not exist",
        "error_code": "NOT_FOUND"
    }
}


---
*End of the test suite. After this: verdicts for Phases 4–5, then the consolidated findings write-up (raw-API findings merged with the SDK-era findings, mapped claim-by-claim) — the backbone of the feedback deliverable for Jitesh.*